doc/specifications/proposed/content_download/content_download.md

```diff
@@ -0,0 +1,60 @@
+# Content download
+
+> Author: bd@ez.no
+
+## Summary
+This feature covers downloading of Content binary files over HTTP.
+
+It covers:
+- generation of proper links from Twig as well as PHP,
+- permissions checking (the user must be allowed to read the Content)
+- streaming of the file over HTTP
+
+## Components
+
+### Twig helper
+
+> Name: ez_download_path
+> Usage: `ez_download_path( content, 'file' )`
+
+The helper will be used to generate a download link for a given Field/Content.
+
+### Controller action + route
+
+#### Route
+
+Path: `/content/download/{contentId}/{fieldIdentifier}/{fileName}`
+
+Example: ``/content/download/68/file/My file.pdf
+
+Query parameters
+
+- version (optional)
+
+  The version number the file must be downloaded for. Requires the versionview permission.
+  If not specified, the published version is used.
```

Comment:
After a face-to-face discussion, we concluded that it was better not to change anything, and keep the version as a query parameter. It was confirmed that copying & communicating the download uri was a common practice, and that it often resulted in obsolete uris in the wild.

*But* we document that the version number *must* be added to the URI to generate reverse proxies compatible uris. A simple helper for that would be to add (and document) an extra, boolean parameter for ez_route, like `include_version: true`.

If necessary, we could also make it a global, semantic option, so that all content/download uris are added the version number.